dialoptions.go

```diff
@@ -89,6 +96,14 @@ type DialOption interface {
 
 var globalDialOptions []DialOption
 
+// LateApplyDialOption takes a parsed target and returns a dial option to apply.
+type LateApplyDialOption interface {
+	// DialOption returns a Dial Option to apply.
+	DialOption(parsedTarget url.URL) DialOption
+}
+
+var globalLateApplyDialOptions []LateApplyDialOption
```

Comment:
Should we ditch this name since they don't have a `lateApply` method?  How about `perTargetDialOption` or something?

Reply:
Alright will switch to perTarget.